src/riscv-unpriv-integration.adoc

```diff
@@ -0,0 +1,388 @@
+[[rv32ch]]
+== RV32CH and RV64CH Base Capability Instruction Sets, Version 1.0
+This chapter describes the RV32CH and RV64CH base capability
+instruction sets, that extend the RV32I and RV64I integer instruction sets with CHERI.
+
+CHERI enhances the base ISA to add hardware memory access control.
+It has an additional memory access mechanism that protects _references to code and data_ (pointers), rather than the _location of code and data_ (integer addresses).
+This mechanism is implemented by providing a new primitive, called a *capability*, that software components can use to implement strongly protected pointers within an address space.
+
+=== CHERI overview
+
+{cheri_base_ext_name} introduces a new data type, the _capability_, generally used for architectural pointers.
+Capabilities are unforgeable and delegatable tokens of authority that grant software the ability to perform a specific set of operations.
+In CHERI, integer-based pointers can be replaced by capabilities to provide memory access control.
+Capability registers extend existing XLEN-bit integer registers to `2*XLEN` (hereafter referred to as CLEN), adding metadata to protect its integrity, limit how it is manipulated, and control its use.
+In addition to widening to CLEN, each capability register also gains a one-bit validity tag.
+
+.CHERI Capability structure
+[#cap_structure]
+[bytefield]
+----
+(defattrs :plain [:plain {:font-family "M+ 1p Fallback" :font-size 25}])
+(def row-height 80)
+(def row-header-fn nil)
+(def boxes-per-row 32)
+(draw-column-headers {:height 50 :font-size 18 :labels (reverse (concat ["0"] (repeat 28 "") ["XLEN-1" "" ""]))})
+
+(draw-box "V"    {:span 1})
+(draw-box "" {:span 1 :borders {}})
+(draw-box "Metadata (bounds, permissions, etc.)"    {:span 30})
+(draw-box "" {:span 2 :borders {}})
+(draw-box "Address"     {:span 30})
+(draw-box "" {:span 2 :borders {}})
+(draw-box "XLEN" {:span 30 :borders {}})
+----
+
+CHERI capabilities consist of:
+
+* An XLEN-bit address, describing where the capability currently points.
+* XLEN bits of metadata:
+** Bounds: a _base_ and a _top_ address, describing the range of addresses the capability can be used to access.
+** Permissions (read, write, execute, ...) describing the kinds of operations the capability can be used for.
+** Sealing information: a capability can be _sealed_, restricting it to only be used or modified in particular ways.
+* A one-bit integrity tag (the `V` field in <<cap_structure>>): this is maintained by hardware and cannot be directly modified by software.
```

Comment:
Krste hated the name "tag" before. here's it's integrity tag, below it's just tag.
`as the tag will be cleared.`

I think we need to choose a good name - maybe in a variable? `Validity tag` perhaps to go with V in the diagram?

Reply:
Updated to use V